sgkit_plink/cli.py

```diff
@@ -0,0 +1,95 @@
+"""
+Command line utilities for conversion.
+"""
+import argparse
+import os
+import signal
+
+from dask.diagnostics import ProgressBar
+
+from . import read_plink
+
+
+def set_sigpipe_handler():
+    if os.name == "posix":
+        # Set signal handler for SIGPIPE to quietly kill the program.
+        signal.signal(signal.SIGPIPE, signal.SIG_DFL)
+
+
+def run_plink2sg(args):
+    # TODO add args for bim and fam seps
+    # TODO add logging and verbosity so we can write out info messages
+    # about what we're doing here.
+    ds = read_plink(args.plink_path, bim_sep=" ", fam_sep=" ")
+
+    # TODO add a command line options for this progess bar.
+    with ProgressBar():
+
+        # TODO Do the dask compute manually so that we can have
+        # more influence over the number of threads used.
+
+        # TODO add options to give more control over the zarr
+        # format used. We probably want to have the option of
+        # writing to a ZipFile anyway, so that we don't have
+        # gazillions of files.
+
+        # TODO catch keyboard interrups here, clean up the
+        # zarr file, and write a meaningful error message.
+        ds.to_zarr(args.sgkit_path, mode="w")
+    # TODO write out an INFO summary of the size of the
+    # input and output
+
+
+def add_plink2sg_arguments(parser):
+
+    # TODO we don't seem to have an __version__ defined yet.
+    # parser.add_argument(
+    #     "-V",
+    #     "--version",
+    #     action="version",
+    #     version=f"%(prog)s {sgkit_plink.__version__}",
+    # )
+
+    parser.add_argument("plink_path", help="The plink dataset to read")
+    parser.add_argument("sgkit_path", help="The path to write the converted dataset to")
+
+
+def get_plink2sg_parser(parser=None):
+
+    parser = argparse.ArgumentParser(description="Convert plink files to sgkit format")
+    add_plink2sg_arguments(parser)
+
+    return parser
+
+
+def get_sgkit_plink_parser():
+    top_parser = argparse.ArgumentParser(
```

Comment:
Do you have any strong opinions on using argparse vs Click or [Fire](https://github.com/google/python-fire)?

Reply:
No, haven't got strong opinions. I haven't tried either of these. How about I get something basic up and running with argparse, and we can weigh up the pros and cons of external packages based on this?

Reply:
There's also [tiangolo/typer](https://github.com/tiangolo/typer) which is not likely to be our solution but is worth knowing about. I like it because it's from the same author as FastAPI so you get a consistent experience across CLI and web Interfaces.